src/runtime/audience-activity-listener-test.js

```diff
@@ -0,0 +1,94 @@
+  } from '../proto/api_messages';
+  import {ClientEventManager} from './client-event-manager';
+  import {AudienceActivityEventListener} from './audience-activity-listener';
+  import {XhrFetcher} from './fetcher';
+  import {ConfiguredRuntime} from './runtime';
+  import {ExperimentFlags} from './experiment-flags';
+  import {PageConfig} from '../model/page-config';
+  import {setExperimentsStringForTesting} from './experiments';
+  
+  describes.realWin('AudienceActivityEventListener', {}, (env) => {
+    let pageConfig;
+    let runtime;
+    let eventManagerCallback;
+    let audienceActivityEventListener;
+    let eventsLoggedToService;
+  
+    const productId = 'pub1:label1';
+  
+    beforeEach(() => {
+        setExperimentsStringForTesting('');
+        eventsLoggedToService = [];
+    sandbox
+      .stub(env.win.document, 'referrer')
+      .get(() => 'https://scenic-2017.appspot.com/landing.html');
+
+    sandbox
+      .stub(XhrFetcher.prototype, 'sendBeacon')
+      .callsFake((unusedUrl, message) => {
+        eventsLoggedToService.push(message);
+      });
+    sandbox
+      .stub(ClientEventManager.prototype, 'registerEventListener')
+      .callsFake((callback) => (eventManagerCallback = callback));
+    pageConfig = new PageConfig(productId);
+    runtime = new ConfiguredRuntime(env.win, pageConfig);
+    audienceActivityEventListener = new AudienceActivityEventListener(runtime, runtime.fetcher_);
+    });
+
+    describe('Audience Activity Events Experiment', () => {
+      
+        it('should not log audience activity events if experiment is off', async () => {
+          // This triggers an event.
+          eventManagerCallback({
+            eventType: AnalyticsEvent.IMPRESSION_PAYWALL,
+            eventOriginator: EventOriginator.UNKNOWN_CLIENT,
+            isFromUserAction: null,
+            additionalParameters: null,
+          });
+  
+          // These wait for the listener to be ready to send data.
+          expect(audienceActivityEventListener.lastAction_).to.not.be.null;
+          await audienceActivityEventListener.lastAction_;
+      
+          expect(eventsLoggedToService.length).to.equal(0);
+        });
+  
+        it('should log audience activity events if experiment is turned on', async () => {
+          setExperimentsStringForTesting(ExperimentFlags.LOGGING_AUDIENCE_ACTIVITY);
+          audienceActivityEventListener.start();
+          // This triggers an event.
+          eventManagerCallback({
+            eventType: AnalyticsEvent.IMPRESSION_PAYWALL,
+            eventOriginator: EventOriginator.UNKNOWN_CLIENT,
+            isFromUserAction: null,
+            additionalParameters: null,
+          });
+
+          // These wait for the listener to be ready to send data.
+          expect(audienceActivityEventListener.lastAction_).to.not.be.null;
+          await audienceActivityEventListener.lastAction_;
+          
+          expect(eventsLoggedToService.length).to.equal(1);
```

Comment:
Can we check the event type to be logged here is equal to IMPRESSION_PAYWALL?
Also, can we add a check for path?

Refer to https://github.com/subscriptions-project/swg-js/blob/main/src/runtime/propensity-server-test.js#L106-L107